### PR TITLE
Add separate rule `exclude` config

### PR DESCRIPTION
**Pull request by incendial**
### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix
[ ] New rule
[ ] Changes an existing rule
[ ] Add autofixing to a rule
[ ] Add a CLI option
[X] Add something to the core
[ ] Other, please explain:

### CODE REVIEW

**Comment by incendial** [comment]
@dkrutskikh I actually don't know how to test it correctly, any suggestions?

File: lib/src/utils/rule_utils.dart [hunk 1]
```diff
@@ -25,13 +27,25 @@ Issue createIssue({
...
       pathSegments: ['dart-code-metrics', 'rules', '$ruleId.html'],
     );
 
-/// Returns the [Severity] from map based [config] otherwise [defaultValue]
+/// Returns a [Severity] from map based [config] otherwise [defaultValue]
 Severity readSeverity(Map<String, Object?> config, Severity defaultValue) =>
     Severity.fromString(config['severity'] as String?) ?? defaultValue;
+
+/// Returns a list of excludes from the given [config]
+Iterable<String> readExcludes(Map<String, Object> config) {
```
Comment (dkrutskikh) [inline lib/src/utils/rule_utils.dart:42]: @incendial please write separate test for this function

File: lib/src/obsoleted/metrics_analyzer.dart [hunk 2]
```diff
@@ -285,8 +290,17 @@ class MetricsAnalyzer {
   Iterable<Issue> _checkOnCodeIssues(
     Suppression ignores,
     InternalResolvedUnitResult source,
+    String filePath,
```
Comment (dkrutskikh) [inline lib/src/obsoleted/metrics_analyzer.dart:293]: @incendial I'd suggest to receive here absolute path

**Review by dkrutskikh — CHANGES_REQUESTED** [review]

**Comment by PiN73** [comment]
Would it be a good option to exclude some pattern from all dart_code_metrics rules (keeping standard dart rules)?
```yaml
dart_code_metrics:
  exclude:
    - lib/**.freezed.dart
```

**Comment by incendial** [comment]
@PiN73 Could you please elaborate on the motivation for you to exclude only Dart Code Metrics rules? It's definitely not a part of this request, but could be implemented separately.

**Comment by PiN73** [comment]
@incendial yes, it would be useful to allow excluding generated files (like .g.dart or .freezed.dart from [freezed](https://pub.dev/packages/freezed)) because I don't control its contents at all

**Comment by incendial** [comment]
@PiN73 okay, sounds reasonable to me, but I don't think that we'd add this feature in 3.1. Could you please create a separate issue for it with the motivation?

File: lib/src/obsoleted/metrics_analyzer.dart (hunk 2 above)
Comment (incendial) [reply lib/src/obsoleted/metrics_analyzer.dart:294]: Discussed it, decided to keep as is for now.

File: lib/src/utils/rule_utils.dart (hunk 1 above)
Comment (incendial) [reply lib/src/utils/rule_utils.dart:44]: Done

**Review by dkrutskikh — APPROVED** [review]

**Pull request merged**